source/uwp/Renderer/lib/AdaptiveChoiceSetInputRenderer.cpp

```diff
@@ -169,6 +169,28 @@ namespace winrt::AdaptiveCards::Rendering::Uwp::implementation
                     // Otherwise, leave all options unset
                     radioButton.IsChecked(IsChoiceSelected(values, input));
                 }
+
+                radioButton.PreviewKeyDown(
+                    [this, stackPanel](winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const& args) -> void
+                    {
+                        std::uint32_t currentIndex;
+                        auto children = stackPanel.Children();
+                        auto size = children.Size();
+                        if (const auto isButtonFound = children.IndexOf(sender.as<winrt::RadioButton>(), currentIndex))
+                        {
+                            if (args.Key() == winrt::VirtualKey::Down)
+                            {
+                                currentIndex = (currentIndex + 1) % size;
+                                children.GetAt(currentIndex).as<winrt::RadioButton>().IsChecked(true);
+                            }
+                            else if (args.Key() == winrt::VirtualKey::Up)
+                            {
+                                currentIndex = (currentIndex + size - 1) % size;
+                                children.GetAt(currentIndex).as<winrt::RadioButton>().IsChecked(true);
+                            }
+                        }
```

Comment:
also, it'd be probably be a good idea to set `args.Handled(true)` since we've handled the event. Unless we want to bubble it up for some reason?

Reply:
Just checked this out and marking this event as handled stops focus functionality :)